### PR TITLE
Docs: Add JS guidelines doc with first few rules explained

### PR DESCRIPTION
**Pull request by spen**
As talked about in #1213, here is a 'first draft' of a JS style guide doc.
I've only added a few explanations of deviations from AirBnB for now - more could be added over time, whether that's before or after a merge.

I realise that I might not be in the best position to really see this through on my own, mostly because my time has been spent working on non RBP projects for some months now 😕
Also, there are eslint rules that I'm not so sure about and so can't write accurate justifications for them.

Happy to hand this over, let it go (if it's not deemed important) or just work with others who know better than I do - what ever works best! 🙂

### CODE REVIEW

**Comment by KarandikarMihir** [comment]
This is looking great, @spen! 😄 👍  Thanks for taking out time for this, we really do appreciate it. You can update the doc one rule at a time, at your own pace. No rush! In the meantime, I'll try to follow your style and add some more rules.

File: docs/js/guidlines.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+While it's good practice to follow a common code style, the most important thing is that you and your team agree on a consistant coding style. The default style guides included here in React Boilerplate are highly recommended, but if your team would like to change some rules - feel free to do so!
+
+## Overrides
+
+### arrow-parens 
+
+eslint: [`arrow-parens`](http://eslint.org/docs/rules/arrow-parens)
+
+> Why? Not sure
```
Comment (mxstbr) [inline docs/js/guidlines.md]: Consistency 😉

**Comment by mxstbr** [comment]
This is awesome, feel free to take your time! (though I can't wait to land this)

File: docs/js/guidlines.md (hunk 1 above)
Comment (spen) [inline docs/js/guidlines.md]: :P I thought that might be the case but wasn't sure!
More specifically, is it for consistency with 2+ arg functions? `( one ) =>` & `( one, two ) =>`?
I think that's a fair comment so I'll update this bit in the morning (it's getting late in Sydney!)

**Comment by spen** [comment]
Thanks for your initial thoughts guys :)
I'm sure it won't take much longer than a week or two anyway - I might just have to throw plenty of Q's about individual rule choices :)

Also, what do you think to addressing guidelines by eslint rule? I was going for that originally but AirBnBs guide tends to talk about a subject, then list related rules... this might be a nicer approach for readability... I'll give that a go in my next batch!

**Comment by gihrig** [comment]
> what do you think to addressing guidelines by eslint rule?

Great idea! Considering that RBP style is AirBnB with exceptions, as listed in `package.json`, documentation and reasoning by rule seems the logical choice. That's where/how RBP devs are likely to encounter the subject.

An overall philosophy regarding these choices would be good as an intro, we already have that from docs/general/README.md [Introduction](https://github.com/react-boilerplate/react-boilerplate/blob/master/docs/general/README.md#introduction):

>Using React Boilerplate, you get to start your app with our community's current ideas on what represents optimal developer experience, best practice, most efficient tooling and cleanest project structure.

**Comment by spen** [comment]
Sorry folks, I've not got enough time to follow up on this especially as it's likely a bit dated by now.
Feel free to commandeer <3

**Pull request closed without merging**